ols/constants.py

```diff
@@ -55,14 +55,26 @@ class QueryValidationMethod(StrEnum):
 DEFAULT_RESPONSE_TOKEN_LIMIT = 500
 MINIMUM_CONTEXT_TOKEN_LIMIT = 1
 
-# Model-specific context window size
+# Provider and Model-specific context window size
 # see https://platform.openai.com/docs/models/gpt-4-turbo-and-gpt-4
 # and https://www.ibm.com/docs/en/cloud-paks/cp-data/4.8.x?topic=models-supported-foundation
 CONTEXT_WINDOW_SIZES = {
-    GRANITE_13B_CHAT_V1: 8192,
-    GRANITE_13B_CHAT_V2: 8192,
-    GPT4_TURBO: 128000,
-    GPT35_TURBO: 16385,
+    PROVIDER_BAM: {
+        GRANITE_13B_CHAT_V1: 8192,
+        GRANITE_13B_CHAT_V2: 8192,
+    },
+    PROVIDER_WATSONX: {
+        GRANITE_13B_CHAT_V1: 8192,
+        GRANITE_13B_CHAT_V2: 8192,
+    },
+    PROVIDER_AZURE_OPENAI: {
+        GPT4_TURBO: 128000,
```

Comment:
this value also isn't correct for azure, here's the full set of limits:

https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#model-summary-table-and-region-availability

![image](https://github.com/openshift/lightspeed-service/assets/4974046/7d0be4a5-d0ff-4800-96df-faf027a871e1)

Reply:
unfortunately the "model version" (e.g. "vision-preview" vs "0125-preview", etc) appears to be something chosen by the user in their azure openai instance setup, not an actual "model id"

see: https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/working-with-models?tabs=powershell

i assume @thoraxe picked a model version when he setup our instance

Reply:
hmm what do you expect there? The table seems to say "128000"

Reply:
the table says anywhere from 8k to 128k depending on which gpt4 "version" is being used.

i guess for now we can assume people are using one of the versions that allows 128k (and our docs should steer them to pick one of those when doing an install) but longer term this will need some more thought.

Reply:
3.5 also appears to vary, it sounds like if @thoraxe updates our model version we can get the 16k context limit instead.

Reply:
ah ok, I was under impression that "our" `GPT4_TURBO` equals to the last three rows, which all have the same 128000  limit